auth/httptransport/httptransport.go

```diff
@@ -171,14 +170,10 @@ func AddAuthorizationMiddleware(client *http.Client, creds *auth.Credentials) er
 			base = http.DefaultTransport
 		}
 	}
-	clientUniverseDomain, err := creds.UniverseDomain(context.Background())
-	if err != nil {
-		return err
-	}
 	client.Transport = &authTransport{
 		creds:                creds,
 		base:                 base,
-		clientUniverseDomain: clientUniverseDomain,
+		clientUniverseDomain: creds.UniverseDomainPropertyProvider(),
```

Comment:
Just call this on the creds obj that is already in the middleware. If you call from the middleware it is already lazy and this does not need to be exposed.

Reply:
Removed, thanks!